node/src/Unison/Runtime/Cryptography.hs

```diff
@@ -0,0 +1,85 @@
+{-# LANGUAGE OverloadedStrings #-}
+{-# Language ScopedTypeVariables #-}
+
+module Unison.Runtime.Cryptography where
+
+import Unison.Cryptography
+import Data.ByteString (ByteString)
+import Data.ByteArray as BA
+import qualified Crypto.Random as R
+
+-- cryptonite
+-- import Crypto.Cipher.Types.Base ()
+-- import Crypto.Cipher.AES.Primitive (AES)
+import qualified Crypto.Cipher.AES as AES
+import Crypto.Cipher.Types
+import Crypto.Error
+
+-- Creates a Unison.Cryptography object specialized to use the noise protocol
+-- (http://noiseprotocol.org/noise.html).
+mkCrypto :: forall cleartext symmetricKey . (ByteArrayAccess cleartext, ByteArray cleartext, ByteArrayAccess symmetricKey, ByteArray symmetricKey) => ByteString -> Cryptography ByteString symmetricKey () () () () cleartext
```

Comment:
Yeah, regardless of what cryptonite does, should make symmetric key a concrete type. If cryptonite already has such a type, great, use it, otherwise define it yourself. It's bad form to just use any ByteString as the key type, as most bytestrings are not valid keys. Providing a newtype with a hidden constructor lets you fail fast at the point that the key is created from a bytestring.